node/node_syncing.go

```diff
@@ -103,6 +107,16 @@ func (node *Node) IsOutOfSync(consensusBlockInfo *consensus.BFTBlockInfo) bool {
 	return true
 }
 
+// DoSync syncs with peers until catchup, this function is not coupled with consensus
+func (node *Node) DoSync() {
+	<-node.peerReadyChan
+	ss := syncing.CreateStateSync(node.SelfPeer.IP, node.SelfPeer.Port, node.Consensus.PubKey.GetAddress())
```

Comment:
If this is not related to consensus, so the node.Consensus.PubKey.GetAddress() shouldn't be needed, right?

Reply:
If the public key is used to identify the peer ip and address, we still need it. Since the synced node will broadcast verified block to unsynced node. Public key is a way to identify which unsynced node to send

Reply:
Node needs to have a public key outside of the `consensusObj`  It can then set the consensus pubkey to be same. This will help us decouple consensus from node identity.